### PR TITLE
Fix compiling issues on illumos

### PR DESCRIPTION
**Pull request by cmouse**

### CODE REVIEW

**Comment by cmouse** [comment]
Tested by @jeffpc, fixes #3028

**Comment by cmouse** [comment]
Ready for review

File: pdns/iputils.hh [hunk 1]
```diff
@@ -59,6 +59,25 @@
 #define le64toh(x) OSSwapLittleToHostInt64(x)
 #endif
 
+#ifdef BE_64
```
Comment (pieterlexis) [inline pdns/iputils.hh]: Could you add a comment about this being for Illumos?

Comment (jeffpc) [inline pdns/iputils.hh]: I suppose you could try to check for some flavor of __sun__, __sun, etc. instead... but (1) I don't remember the magic define, and (2) it's ugly anyway.

**Pull request merged**